clang/test/SemaSYCL/nested-anon-and-std-ns.cpp

```diff
@@ -0,0 +1,53 @@
+// RUN: %clang_cc1 -fsycl -fsycl-is-device -internal-isystem %S/Inputs -fsyntax-only -sycl-std=2020 -verify %s
+
+#include "sycl.hpp"
+
+namespace std {
+namespace NestedInStd {
+struct NestedStruct {};
+}; // namespace NestedInStd
+}; // namespace std
+
+namespace {
+namespace NestedInAnon {
+struct StructInAnonymousNS {};
+} // namespace NestedInAnon
+
+} // namespace
+
+namespace ValidNS {
+struct StructinValidNS {};
+} // namespace ValidNS
+
+struct MyWrapper {
+
+public:
+  void test() {
+    cl::sycl::queue q;
+
+    // expected-error@Inputs/sycl.hpp:220 {{'std::NestedInStd::NestedStruct' is an invalid kernel name type}}
+    // expected-note@Inputs/sycl.hpp:220 {{'std::NestedInStd::NestedStruct' cannot be in the "std" namespace}}
```

Comment:
Drive-by nit with the existing diagnostic wording: `cannot be in the "std" namespace` should be `cannot be in the 'std' namespace` (with single quotes instead of double -- it's a syntactic identifier).